cpp/src/io/statistics/column_stats.cu

```diff
@@ -481,52 +471,40 @@ void __device__ mergeIntColumnStats(merge_state_s *s,
     non_nulls += ck->non_nulls;
     null_count += ck->null_count;
   }
-  non_nulls = cub::WarpReduce<uint32_t>(storage.u32[t / 32]).Sum(non_nulls);
-  __syncwarp();
   vmin = cub::WarpReduce<int64_t>(storage.i64[t / 32]).Reduce(vmin, cub::Min());
   __syncwarp();
   vmin = shuffle(vmin);
-
-  null_count = cub::WarpReduce<uint32_t>(storage.u32[t / 32]).Sum(null_count);
-  __syncwarp();
   vmax = cub::WarpReduce<int64_t>(storage.i64[t / 32]).Reduce(vmax, cub::Max());
   __syncwarp();
   vmax = shuffle(vmax);
-
-  vsum = cub::WarpReduce<int64_t>(storage.i64[t / 32]).Sum(vsum);
-
   if (!(t & 0x1f)) {
-    s->warp_non_nulls[t >> 5] = non_nulls;
-    s->warp_nulls[t >> 5]     = null_count;
     s->warp_min[t >> 5].i_val = vmin;
     s->warp_max[t >> 5].i_val = vmax;
-    s->warp_sum[t >> 5].i_val = vsum;
   }
   has_minmax = __syncthreads_or(vmin <= vmax);
+  non_nulls  = cub::BlockReduce<uint32_t, block_size>(storage.block_u32).Sum(non_nulls);
+  __syncthreads();
+  null_count = cub::BlockReduce<uint32_t, block_size>(storage.block_u32).Sum(null_count);
+  __syncthreads();
+  vsum = cub::BlockReduce<int64_t, block_size>(storage.block_i64).Sum(vsum);
+  __syncthreads();
+
   if (t < 32 * 1) {
     vmin = cub::WarpReduce<int64_t>(storage.i64[t / 32]).Reduce(s->warp_min[t].i_val, cub::Min());
     if (!(t & 0x1f)) {
       s->ck.min_value.i_val = vmin;
       s->ck.has_minmax      = (has_minmax);
+      s->ck.non_nulls       = non_nulls;
+      s->ck.null_count      = null_count;
```

Comment:
So, in theory it should be faster to let a thread from a different warp write `non_nulls` and `null_count`. I'm not sure if this is relevant at all.

Reply:
I can revert the changes as earlier, I felt the code was too spread out.